haskell/cabal
# `check`: add PackageInfo warning

“Backport” of #9481, should be the last step (and close) #9331.

Include the following checklist in your PR:


---

**QA instructions**

- create a package
- use the PackageInfo_* autogenerated module
- `cabal check` will warn and suggest you to update to `cabal` 3.12.

---

Things to double-check:
- the logic behind the check (it fires every time it finds a `PackageInfo_*` module, regardless of `cabal-version`)
- the warning message
  ```
  Warning: To use the autogenerated module PackageInfo_* you need to specify `cabal-version: 3.12` or higher. This is not possible in `cabal 3.10`. To use this feature and be able to upload your package to Hackage, download `cabal 3.12`!
  ```
  Note that 3.12 has not been released yet.
- does this actually close #9331?
